Index: ubuntu/debian/compiz-manager
===================================================================
--- ubuntu.orig/debian/compiz-manager	2008-03-12 14:44:20.000000000 +0100
+++ ubuntu/debian/compiz-manager	2008-03-12 14:44:25.000000000 +0100
@@ -379,18 +379,6 @@
 	verbose "Dry run finished: everything should work with regards to Compiz and 3D.\n"
 	exit 0;
 fi
-# start the gtk-window-decorator if present
-if [ -x ${COMPIZ_BIN_PATH}emerald ] && [ "$USE_EMERALD" = "yes" ]; then
-	verbose "Starting emerald\n"
-	${COMPIZ_BIN_PATH}emerald --replace &
-elif [ -x ${COMPIZ_BIN_PATH}gtk-window-decorator ] && [ -n "$GNOME_DESKTOP_SESSION_ID" ]; then
-	verbose "Starting gtk-window-decorator\n"
-	${COMPIZ_BIN_PATH}gtk-window-decorator --replace &
-elif [ -x ${COMPIZ_BIN_PATH}kde-window-decorator ] && [ -n "$KDE_FULL_SESSION" ]; then
-	verbose "Starting kde-window-decorator\n"
-	${COMPIZ_BIN_PATH}kde-window-decorator --replace &
-	FALLBACKWM="${KWIN}"
-fi
 
 ${COMPIZ_BIN_PATH}${COMPIZ_NAME} $COMPIZ_OPTIONS "$@" $COMPIZ_PLUGINS || exec $FALLBACKWM $FALLBACKWM_OPTIONS
 
