--- 0-compiz.orig/debian/compiz-manager
+++ 0-compiz/debian/compiz-manager
@@ -118,6 +118,21 @@ abort_with_fallback_wm()
 	fi
 }
 
+# Check if we run with the Software Rasterizer, this happens e.g.
+# when a second screen session is opened via f-u-a on intel
+check_software_rasterizer()
+{
+	verbose "Checking for Software Rasterizer: "
+	if glxinfo 2> /dev/null | egrep -q '^OpenGL renderer string: Software Rasterizer' ; then
+		verbose "present. \n";
+		return 0;
+	else
+		verbose "Not present. \n"
+		return 1;
+	fi
+
+}
+
 # Check for non power of two texture support
 check_npot_texture()
 {
@@ -389,6 +404,12 @@ if ! check_xgl; then
 		abort_with_fallback_wm
 	fi
 
+	# check if we run with software rasterizer and if so, bail out
+	if check_software_rasterizer; then
+		verbose "Software rasterizer detected, aborting"
+		abort_with_fallback_wm
+	fi
+
 	if check_nvidia && ! check_nvidia_memory; then
 		abort_with_fallback_wm
 	fi
