Index: compiz-0.6.99+git20071106/configure
===================================================================
--- compiz-0.6.99+git20071106.orig/configure	2007-11-06 11:05:59.000000000 -0600
+++ compiz-0.6.99+git20071106/configure	2007-11-06 11:19:55.000000000 -0600
@@ -22998,6 +22998,7 @@
 		 xdamage    \
 		 xrandr	    \
 		 xinerama   \
+		 xcursor    \
 		 ice	    \
 		 sm	    \
 		 libxml-2.0 \
