commit 0e02a98ade9aeb1d374a46931e7fdc735a7e750d
Author: David Reveman <davidr@novell.com>
Date:   Thu May 31 11:54:53 2007 -0400

    Fix byte order check in imageBufferToTexture.

diff --git a/src/texture.c b/src/texture.c
index 384ae3d..bd0be59 100644
--- a/src/texture.c
+++ b/src/texture.c
@@ -166,10 +166,10 @@ imageBufferToTexture (CompScreen   *screen,
 {
 #if IMAGE_BYTE_ORDER == MSBFirst
     return imageToTexture (screen, texture, image, width, height,
-			   GL_BGRA, GL_UNSIGNED_BYTE);
+			   GL_BGRA, GL_UNSIGNED_INT_8_8_8_8_REV);
 #else
     return imageToTexture (screen, texture, image, width, height,
-			   GL_BGRA, GL_UNSIGNED_INT_8_8_8_8_REV);
+			   GL_BGRA, GL_UNSIGNED_BYTE);
 #endif
 }
 
