Description: Miscellaneous fixes and workarounds from Ubuntu.

--- a/metadata/core.xml.in
+++ b/metadata/core.xml.in
@@ -139,7 +139,7 @@
 			<option name="focus_prevention_match" type="match">
 			<_short>Focus Prevention Windows</_short>
 			<_long>Focus prevention windows</_long>
-			<default>any</default>
+			<default>!(class=Polkit-gnome-authentication-agent-1)</default>
 			</option>
 			<option type="bool" name="remember_vp_focus">
 			<_short>Remember Focus</_short>
--- a/plugins/fade/fade.xml.in
+++ b/plugins/fade/fade.xml.in
@@ -10,6 +10,7 @@
 	    </requirement>
 	    <relation type="after">
 		<plugin>decor</plugin>
+		<plugin>staticswitcher</plugin>
 	    </relation>
 	    <relation type="before">
 		<plugin>cube</plugin>
@@ -50,7 +51,7 @@
 	    <option name="window_match" type="match">
 		<_short>Fade windows</_short>
 		<_long>Windows that should be fading</_long>
-		<default>any</default>
+                <default>any &amp; !(title=notify-osd)</default>
 	    </option>
 	    <option name="visual_bell" type="bell">
 		<_short>Visual Bell</_short>
--- a/plugins/gnomecompat/gnomecompat.xml.in
+++ b/plugins/gnomecompat/gnomecompat.xml.in
@@ -54,6 +54,7 @@
 		<option name="run_command_terminal_key" type="key">
 		    <_short>Open a terminal</_short>
 		    <_long>Open a terminal</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;T</default>
 		</option>
 	    </group>
 	</options>
