Index: compiz/plugins/compiz.schemas.in
===================================================================
--- compiz.orig/plugins/compiz.schemas.in	2006-09-29 12:46:33.000000000 +0800
+++ compiz/plugins/compiz.schemas.in	2006-09-29 12:46:39.000000000 +0800
@@ -24608,5 +24608,933 @@
             </locale>
         </schema>
 
+        <!-- plane options -->
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/info</key>
+            <applyto>/apps/compiz/plugins/plane/info</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default></default>
+            <locale name="C">
+                <short>Desktop Plane</short>
+                <long>Place windows on a plane</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/load_before</key>
+            <applyto>/apps/compiz/plugins/plane/load_before</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plugins that this must load before</short>
+                <long>Do not modify</long>
+            </locale>
+            <locale name="cs">
+                <short>Moduly, které musí být tímto předem načteny</short>
+                <long>Neměnit</long>
+            </locale>
+            <locale name="de">
+                <short>Plugins, die zuvor geladen werden müssen</short>
+                <long>Nicht ändern</long>
+            </locale>
+            <locale name="es">
+                <short>Complementos que esto debe cargar antes</short>
+                <long>No modificar</long>
+            </locale>
+            <locale name="fi">
+                <short>Etukäteen tätä varten ladattavat liitännäiset</short>
+                <long>Älä muuta</long>
+            </locale>
+            <locale name="fr">
+                <short>Plug-ins à charger avant</short>
+                <long>Ne pas modifier</long>
+            </locale>
+            <locale name="hu">
+                <short>Kötelezően betöltendő beépülő modulok </short>
+                <long>Nem módosít</long>
+            </locale>
+            <locale name="it">
+                <short>Plug-in da caricare per primi</short>
+                <long>Non modificare</long>
+            </locale>
+            <locale name="ja">
+                <short>これを前にロードする必要のあるプラグイン</short>
+                <long>変更しないでください</long>
+            </locale>
+            <locale name="pl">
+                <short>Wtyczki, które muszą zostać załadowane przed</short>
+                <long>Bez zmian</long>
+            </locale>
+            <locale name="pt_BR">
+                <short>Plug-ins que devem ser carregados antes de</short>
+                <long>Não modificar</long>
+            </locale>
+            <locale name="sv">
+                <short>Insticksmoduler som måste läsas in före</short>
+                <long>Ändra inte</long>
+            </locale>
+            <locale name="zh_CN">
+                <short>必须先装载的插件</short>
+                <long>不修改</long>
+            </locale>
+            <locale name="zh_TW">
+                <short>在此之前必須先載入的外掛程式</short>
+                <long>不要修改</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/requires</key>
+            <applyto>/apps/compiz/plugins/plane/requires</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plugins that this requires</short>
+                <long>Do not modify</long>
+            </locale>
+            <locale name="cs">
+                <short>Moduly, které jsou tímto vyžadovány</short>
+                <long>Neměnit</long>
+            </locale>
+            <locale name="de">
+                <short>Erforderliche Plugins</short>
+                <long>Nicht ändern</long>
+            </locale>
+            <locale name="es">
+                <short>Complementos que esto requiere</short>
+                <long>No modificar</long>
+            </locale>
+            <locale name="fi">
+                <short>Tähän vaadittavat liitännäiset</short>
+                <long>Älä muuta</long>
+            </locale>
+            <locale name="fr">
+                <short>Plug-ins requis</short>
+                <long>Ne pas modifier</long>
+            </locale>
+            <locale name="hu">
+                <short>Szükséges beépülő modulok</short>
+                <long>Nem módosít</long>
+            </locale>
+            <locale name="it">
+                <short>Plug-in necessari</short>
+                <long>Non modificare</long>
+            </locale>
+            <locale name="ja">
+                <short>これが必要なプラグイン</short>
+                <long>変更しないでください</long>
+            </locale>
+            <locale name="pl">
+                <short>Wtyczki wymagane przez ten element</short>
+                <long>Bez zmian</long>
+            </locale>
+            <locale name="pt_BR">
+                <short>Plug-ins necessários</short>
+                <long>Não modificar</long>
+            </locale>
+            <locale name="sv">
+                <short>Insticksmoduler som den här kräver</short>
+                <long>Ändra inte</long>
+            </locale>
+            <locale name="zh_CN">
+                <short>需要的插件</short>
+                <long>不修改</long>
+            </locale>
+            <locale name="zh_TW">
+                <short>這時需要的外掛程式</short>
+                <long>不要修改</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_left_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_left_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>&lt;Control&gt;&lt;Alt&gt;Left</default>
+            <locale name="C">
+                <short>Plane Left</short>
+                <long>Plane left</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_left_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_left_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane Left</short>
+                <long>Plane left</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_left_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_left_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane Left</short>
+                <long>Plane left</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_left_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_left_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane Left</short>
+                <long>Plane left (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_right_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_right_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>&lt;Control&gt;&lt;Alt&gt;Right</default>
+            <locale name="C">
+                <short>Plane Right</short>
+                <long>Plane right</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_right_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_right_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane Right</short>
+                <long>Plane right</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_right_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_right_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane Right</short>
+                <long>Plane right</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_right_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_right_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane Right</short>
+                <long>Plane right (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_down_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_down_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>&lt;Control&gt;&lt;Alt&gt;Down</default>
+            <locale name="C">
+                <short>Plane Down</short>
+                <long>Plane down</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_down_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_down_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane Down</short>
+                <long>Plane down</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_down_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_down_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane Down</short>
+                <long>Plane down</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_down_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_down_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane Down</short>
+                <long>Plane down (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_up_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_up_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>&lt;Control&gt;&lt;Alt&gt;Up</default>
+            <locale name="C">
+                <short>Plane Up</short>
+                <long>Plane up</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_up_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_up_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane Up</short>
+                <long>Plane up</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_up_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_up_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane Up</short>
+                <long>Plane up</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_up_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_up_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane Up</short>
+                <long>Plane up (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_1_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_1_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 1</short>
+                <long>Plane to face 1</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_1_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_1_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 1</short>
+                <long>Plane to face 1</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_1_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_1_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 1</short>
+                <long>Plane to face 1</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_1_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_1_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 1</short>
+                <long>Plane to face 1 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_2_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_2_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 2</short>
+                <long>Plane to face 2</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_2_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_2_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 2</short>
+                <long>Plane to face 2</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_2_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_2_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 2</short>
+                <long>Plane to face 2</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_2_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_2_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 2</short>
+                <long>Plane to face 2 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_3_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_3_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 3</short>
+                <long>Plane to face 3</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_3_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_3_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 3</short>
+                <long>Plane to face 3</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_3_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_3_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 3</short>
+                <long>Plane to face 3</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_3_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_3_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 3</short>
+                <long>Plane to face 3 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_4_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_4_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 4</short>
+                <long>Plane to face 4</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_4_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_4_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 4</short>
+                <long>Plane to face 4</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_4_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_4_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 4</short>
+                <long>Plane to face 4</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_4_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_4_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 4</short>
+                <long>Plane to face 4 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_5_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_5_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 5</short>
+                <long>Plane to face 5</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_5_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_5_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 5</short>
+                <long>Plane to face 5</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_5_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_5_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 5</short>
+                <long>Plane to face 5</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_5_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_5_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 5</short>
+                <long>Plane to face 5 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_6_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_6_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 6</short>
+                <long>Plane to face 6</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_6_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_6_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 6</short>
+                <long>Plane to face 6</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_6_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_6_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 6</short>
+                <long>Plane to face 6</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_6_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_6_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 6</short>
+                <long>Plane to face 6 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_7_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_7_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 7</short>
+                <long>Plane to face 7</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_7_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_7_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 7</short>
+                <long>Plane to face 7</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_7_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_7_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 7</short>
+                <long>Plane to face 7</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_7_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_7_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 7</short>
+                <long>Plane to face 7 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_8_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_8_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 8</short>
+                <long>Plane to face 8</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_8_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_8_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 8</short>
+                <long>Plane to face 8</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_8_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_8_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 8</short>
+                <long>Plane to face 8</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_8_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_8_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 8</short>
+                <long>Plane to face 8 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_9_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_9_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 9</short>
+                <long>Plane to face 9</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_9_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_9_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 9</short>
+                <long>Plane to face 9</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_9_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_9_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 9</short>
+                <long>Plane to face 9</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_9_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_9_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 9</short>
+                <long>Plane to face 9 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_10_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_10_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 10</short>
+                <long>Plane to face 10</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_10_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_10_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 10</short>
+                <long>Plane to face 10</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_10_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_10_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 10</short>
+                <long>Plane to face 10</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_10_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_10_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 10</short>
+                <long>Plane to face 10 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_11_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_11_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 11</short>
+                <long>Plane to face 11</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_11_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_11_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 11</short>
+                <long>Plane to face 11</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_11_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_11_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 11</short>
+                <long>Plane to face 11</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_11_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_11_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 11</short>
+                <long>Plane to face 11 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_12_key</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_12_key</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 12</short>
+                <long>Plane to face 12</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_12_button</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_12_button</applyto>
+            <owner>compiz</owner>
+            <type>string</type>
+            <default>Disabled</default>
+            <locale name="C">
+                <short>Plane To Face 12</short>
+                <long>Plane to face 12</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_12_bell</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_12_bell</applyto>
+            <owner>compiz</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Plane To Face 12</short>
+                <long>Plane to face 12</long>
+            </locale>
+        </schema>
+
+        <schema>
+            <key>/schemas/apps/compiz/plugins/plane/allscreens/options/plane_to_12_edge</key>
+            <applyto>/apps/compiz/plugins/plane/allscreens/options/plane_to_12_edge</applyto>
+            <owner>compiz</owner>
+            <type>list</type>
+            <list_type>string</list_type>
+            <default>[]</default>
+            <locale name="C">
+                <short>Plane To Face 12</short>
+                <long>Plane to face 12 (Left, Right, Top, Bottom, TopLeft, TopRight, BottomLeft, BottomRight)</long>
+            </locale>
+        </schema>
+
     </schemalist>
 </gconfschemafile>
