Index: plugins/compiz.schemas.in
===================================================================
--- plugins/compiz.schemas.in	(revision 3541)
+++ plugins/compiz.schemas.in	(working copy)
@@ -10409,7 +10409,7 @@
             <owner>compiz</owner>
             <type>list</type>
             <list_type>string</list_type>
-            <default>[freedektop.png]</default>
+            <default>[freedesktop.png]</default>
             <locale name="C">
                 <short>Image files</short>
                 <long>List of PNG and SVG files that should be rendered on top face of cube</long>
