Index: gtk/window-decorator/gtk-window-decorator.c
===================================================================
--- a/gtk/window-decorator/gtk-window-decorator.c
+++ b/gtk/window-decorator/gtk-window-decorator.c
@@ -5846,6 +5846,8 @@
 	if (theme)
 	{
 	    meta_theme_set_current (theme, TRUE);
+	    if (!meta_theme_get_current())
+	        use_meta_theme = FALSE;
 
 	    g_free (theme);
 	}
