Index: compiz/include/compiz.h
===================================================================
--- compiz.orig/include/compiz.h	2007-12-22 11:13:38.000000000 +0100
+++ compiz/include/compiz.h	2007-12-22 11:13:49.000000000 +0100
@@ -226,6 +226,7 @@
 extern Bool       noDetection;
 extern Bool	  useDesktopHints;
 extern Bool       onlyCurrentScreen;
+extern Bool       noFBO;
 
 extern int  defaultRefreshRate;
 extern char *defaultTextureFilter;
Index: compiz/src/main.c
===================================================================
--- compiz.orig/src/main.c	2007-12-22 11:13:38.000000000 +0100
+++ compiz/src/main.c	2007-12-22 11:13:49.000000000 +0100
@@ -62,6 +62,7 @@
 Bool noDetection = FALSE;
 Bool useDesktopHints = TRUE;
 Bool onlyCurrentScreen = FALSE;
+Bool noFBO = FALSE;
 
 #ifdef USE_COW
 Bool useCow = TRUE;
@@ -83,6 +84,7 @@
 	    "[--sm-disable] "
 	    "[--sm-client-id ID] "
 	    "[--no-detection]\n       "
+	    "[--no-fbo] "
 	    "[--ignore-desktop-hints] "
 	    "[--only-current-screen]"
 
@@ -336,6 +338,10 @@
 	{
 	    onlyCurrentScreen = TRUE;
 	}
+	else if (!strcmp (argv[i], "--no-fbo"))
+	{
+		noFBO = TRUE;
+	}
 
 #ifdef USE_COW
 	else if (!strcmp (argv[i], "--use-root-window"))
Index: compiz/src/screen.c
===================================================================
--- compiz.orig/src/screen.c	2007-12-22 11:13:38.000000000 +0100
+++ compiz/src/screen.c	2007-12-22 11:13:49.000000000 +0100
@@ -1877,7 +1877,7 @@
     }
 
     s->fbo = 0;
-    if (strstr (glExtensions, "GL_EXT_framebuffer_object"))
+    if (!noFBO && strstr (glExtensions, "GL_EXT_framebuffer_object"))
     {
 	s->genFramebuffers = (GLGenFramebuffersProc)
 	    getProcAddress (s, "glGenFramebuffersEXT");
