Index: compiz/include/compiz-core.h
===================================================================
--- compiz.orig/include/compiz-core.h
+++ compiz/include/compiz-core.h
@@ -219,6 +219,7 @@ extern Bool       useCow;
 extern Bool       noDetection;
 extern Bool	  useDesktopHints;
 extern Bool       onlyCurrentScreen;
+extern Bool       noFBO;
 
 extern int  defaultRefreshRate;
 extern char *defaultTextureFilter;
Index: compiz/src/main.c
===================================================================
--- compiz.orig/src/main.c
+++ compiz/src/main.c
@@ -62,6 +62,7 @@ Bool strictBinding = TRUE;
 Bool noDetection = FALSE;
 Bool useDesktopHints = TRUE;
 Bool onlyCurrentScreen = FALSE;
+Bool noFBO = FALSE;
 
 #ifdef USE_COW
 Bool useCow = TRUE;
@@ -83,6 +84,7 @@ usage (void)
 	    "[--sm-disable] "
 	    "[--sm-client-id ID] "
 	    "[--no-detection]\n       "
+	    "[--no-fbo] "
 	    "[--ignore-desktop-hints] "
 	    "[--only-current-screen]"
 
@@ -329,6 +331,10 @@ main (int argc, char **argv)
 	{
 	    onlyCurrentScreen = TRUE;
 	}
+	else if (!strcmp (argv[i], "--no-fbo"))
+	{
+		noFBO = TRUE;
+	}
 
 #ifdef USE_COW
 	else if (!strcmp (argv[i], "--use-root-window"))
Index: compiz/src/screen.c
===================================================================
--- compiz.orig/src/screen.c
+++ compiz/src/screen.c
@@ -2037,7 +2037,7 @@ addScreen (CompDisplay *display,
     }
 
     s->fbo = 0;
-    if (strstr (glExtensions, "GL_EXT_framebuffer_object"))
+    if (!noFBO && strstr (glExtensions, "GL_EXT_framebuffer_object"))
     {
 	s->genFramebuffers = (GLGenFramebuffersProc)
 	    getProcAddress (s, "glGenFramebuffersEXT");
