Description: Fix exec_prefix entry.
Author: Miguel A. Colón Vélez <debian.micove@gmail.com>
Forwarded: no
Last-Update: 2012-03-10

--- a/plugins/compiztoolbox/compiz-compiztoolbox.pc.in
+++ b/plugins/compiztoolbox/compiz-compiztoolbox.pc.in
@@ -1,5 +1,5 @@
 prefix=@prefix@
-exec_prefix=@prefix@
+exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 
--- a/plugins/animationaddon/compiz-animationaddon.pc.in
+++ b/plugins/animationaddon/compiz-animationaddon.pc.in
@@ -1,5 +1,5 @@
 prefix=@prefix@
-exec_prefix=@prefix@
+exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 
