=== modified file 'plugins/opengl/src/paint.cpp'
--- a/plugins/opengl/src/paint.cpp	2012-02-27 03:53:45 +0000
+++ b/plugins/opengl/src/paint.cpp	2012-03-05 09:38:48 +0000
@@ -218,9 +218,6 @@
     CompPoint     offXY;
     int           dontcare;
 
-    CompWindowList                   pl;
-    CompWindowList::reverse_iterator rit;
-
     unredirectFS = CompositeScreen::get (screen)->
 	getOption ("unredirect_fullscreen_windows")->value ().b ();
 
@@ -235,7 +232,8 @@
 	count	       = 0;
     }
 
-    pl = cScreen->getWindowPaintList ();
+    const CompWindowList &pl = cScreen->getWindowPaintList ();
+    CompWindowList::const_reverse_iterator rit;
 
     if (!(mask & PAINT_SCREEN_NO_OCCLUSION_DETECTION_MASK))
     {

