--- 0-compiz.orig/debian/compiz-manager
+++ 0-compiz/debian/compiz-manager
@@ -197,6 +197,10 @@ check_xgl()
 # Check if the nVidia card has enough video ram to make sense
 check_nvidia_memory()
 {
+	if ! which $NVIDIA_SETTINGS >/dev/null 2>&1; then
+		verbose "No $NVIDIA_SETTINGS program available to determine VideoRam."
+		return 0
+	fi
 	MEM=$(${NVIDIA_SETTINGS} -q VideoRam | egrep Attribute\ \'VideoRam\'\ .*: | cut -d: -f3 | sed 's/[^0-9]//g')
 	if [ $MEM -lt $NVIDIA_MEMORY ]; then
 		verbose "Less than ${NVIDIA_MEMORY}kb of memory and nVidia";
