--- 0-compiz.orig/debian/compiz-manager
+++ 0-compiz/debian/compiz-manager
@@ -27,13 +27,13 @@
 # 
 
 
-COMPIZ_BIN_PATH="/usr/local/bin/" # For window decorators and compiz
-PLUGIN_PATH="/usr/local/lib/compiz/" 
+COMPIZ_BIN_PATH="/usr/bin/" # For window decorators and compiz
+PLUGIN_PATH="/usr/lib/compiz/" 
 GLXINFO="/usr/bin/glxinfo"
 KWIN="/usr/bin/kwin"
 METACITY="/usr/bin/metacity"
 XFWM="/usr/bin/xfwm"
-COMPIZ_NAME="compiz" # Final name for compiz (compiz.real) 
+COMPIZ_NAME="compiz.real" # Final name for compiz (compiz.real) 
 
 # For Xgl LD_PRELOAD
 LIBGL_NVIDIA="/usr/lib/nvidia/libGL.so.1.2.xlibmesa"
