imported from upstream commit aed97c441881d9c382c7865d0305fc8f884c10ac
fixes #483170
--- 0-compiz.orig/src/screen.c
+++ 0-compiz/src/screen.c
@@ -2420,6 +2420,13 @@ damageScreenRegion (CompScreen *screen,
     XUnionRegion (screen->damage, region, screen->damage);
 
     screen->damageMask |= COMP_SCREEN_DAMAGE_REGION_MASK;
+
+    /* if the number of damage rectangles grows two much between repaints,
+       we have a lot of overhead just for doing the damage tracking -
+       in order to make sure we're not having too much overhead, damage
+       the whole screen if we have a lot of damage rects */
+    if (screen->damage->numRects > 100)
+	damageScreen (screen);
 }
 
 void
